pkg/scheduler/core/equivalence_cache_test.go

```diff
@@ -689,31 +697,40 @@ func TestInvalidateCachedPredicateItemOfAllNodes(t *testing.T) {
 	ecache := NewEquivalenceCache()
 
 	for _, test := range tests {
-		node := schedulercache.NewNodeInfo()
-		node.SetNode(&v1.Node{ObjectMeta: metav1.ObjectMeta{Name: test.nodeName}})
-		// set cached item to equivalence cache
-		ecache.updateResult(
-			test.podName,
-			testPredicate,
-			test.cachedItem.fit,
-			test.cachedItem.reasons,
-			test.equivalenceHashForUpdatePredicate,
-			test.cache,
-			node,
-		)
+		t.Run(test.name, func(t *testing.T) {
```

Comment:
Given the nature of this test, I think we should actually run this first loop in the main test, and use t.Run in the second loop like you are already doing.

My reasoning is that this test uses the same ecache object across all test cases. (I think that should be fixed up in a separate PR.) so this loop feels more like setup than the actual test.

Reply:
I agree with @misterikkit. You shouldn't `t.Run` in this loop.